components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/dao/impl/ActionManagementDAOImpl.java

```diff
@@ -18,182 +18,141 @@
-                        .encryptAssociatedSecrets(action.getEndpoint().getAuthentication(), actionId);
-
-                // Add Endpoint configuration properties.
-                addEndpointProperties(actionId, getEndpointProperties(action.getEndpoint().getUri(),
-                        action.getEndpoint().getAuthentication().getType().name(), encryptedAuthProperties), tenantId);
-
-                return null;
-            });
-
-            return getActionByActionId(actionType, actionId, tenantId);
-        } catch (TransactionException e) {
-            if (LOG.isDebugEnabled()) {
-                LOG.debug(String.format("Error while creating the Action of Action Type: %s in Tenant Domain: %s." +
-                                " Rolling back created action information and deleting added secrets.", actionType,
-                        IdentityTenantUtil.getTenantDomain(tenantId)));
+    public List<ActionDTO> getActionsByActionType(String actionType, Integer tenantId) throws ActionMgtException {
+
+        List<ActionDTO> actionDTOS = new ArrayList<>();
+        try (Connection dbConnection = IdentityDatabaseUtil.getDBConnection(false);
+             NamedPreparedStatement statement = new NamedPreparedStatement(dbConnection,
+                     ActionMgtSQLConstants.Query.GET_ACTIONS_BASIC_INFO_BY_ACTION_TYPE)) {
+
+            statement.setString(ActionMgtSQLConstants.Column.ACTION_TYPE, actionType);
+            statement.setInt(ActionMgtSQLConstants.Column.TENANT_ID, tenantId);
+
+            try (ResultSet rs = statement.executeQuery()) {
+                while (rs.next()) {
+                    String actionId = rs.getString(ActionMgtSQLConstants.Column.ACTION_UUID);
+
+                    ActionDTO.Builder actionBuilder = new ActionDTO.Builder()
+                            .id(actionId)
+                            .type(org.wso2.carbon.identity.action.management.model.Action.ActionTypes.valueOf(
+                                    rs.getString(ActionMgtSQLConstants.Column.ACTION_TYPE)))
+                            .name(rs.getString(ActionMgtSQLConstants.Column.ACTION_NAME))
+                            .description(rs.getString(ActionMgtSQLConstants.Column.ACTION_DESCRIPTION))
+                            .status(org.wso2.carbon.identity.action.management.model.Action.Status.valueOf(
+                                    rs.getString(ActionMgtSQLConstants.Column.ACTION_STATUS)));
+                    actionBuilder.setEndpointAndProperties(getActionPropertiesFromDB(actionId, tenantId));
+
+                    actionDTOS.add(actionBuilder.build());
+                }
             }
-            throw ActionManagementUtil.handleServerException(
-                    ActionMgtConstants.ErrorMessages.ERROR_WHILE_ADDING_ACTION, e);
+
+            return actionDTOS;
+        } catch (SQLException e) {
+            throw new ActionMgtServerException("Error while retrieving Actions Basic information by Action Type.", e);
```

Comment:
This is not basic info right ? This seems to be populating the full object

Reply:
fixed with https://github.com/wso2/carbon-identity-framework/pull/6119/commits/edeb975693380b404a45f24842338951d17c1695